src/test/java/com/artipie/rpm/RpmITCase.java

```diff
@@ -111,18 +110,44 @@ void tearDown() throws Exception {
     }
 
     @Test
-    void generatesMetadata() {
+    void generatesMetadata() throws Exception {
         new Rpm(this.storage, StandardNamingPolicy.SHA1, Digest.SHA256, true)
             .batchUpdate(Key.ROOT)
             .blockingAwait();
+        final BlockingStorage bsto = new BlockingStorage(this.storage);
+        for (
+            final XmlPackage pckg : new XmlPackage.Stream(true).get().collect(Collectors.toList())
+        ) {
+            final Path xml = this.metadataFromStorage(bsto, pckg);
+            MatcherAssert.assertThat(
+                String.format(
+                    "Metadata %s has %d rpm packages", pckg.name(), RpmITCase.SIZE.count()
+                ),
+                new XMLDocument(xml),
+                new NodeHasPkgCount(RpmITCase.SIZE.count(), pckg.tag())
+            );
+        }
```

Comment:
@olenagerasimova that is quite large chunk of code duplicated in two tests, do I'd suggest to extract a method or a matcher class from it

Reply:
@olegmoz eventually I'm planning to get rid of `batchUpdateIncrementally` method, so these duplications also should be removed